packages/colours/_colours.scss

```diff
@@ -17,7 +17,7 @@ $color-white: #fff;
 $color-gainsboro: #d8d8d8;
 
 // Typography colors
-$color-link: $color-shuttle-grey;
+$color-link: $color-shark;
```

Comment:
Hmmmmm. Is this maybe a lazy way of doing this?

Maybe the `$color-link` variable is just no longer needed and can be deleted?

Maybe the Link component should just use the `$color-text` token in "Link.modules.scss"?

Thoughts?

Reply:
Yeah, I thought about this. Removing it would mean we would have to update the `Link` component as well. Plus keeping it allows us to easily change the link colour in the future. It doesn't have any impact on the way the styles actually compile, so leaving it seems fairly harmless.

Reply:
I see these variables as a form of documentation since it adds verbosity to otherwise generic colours, just like how we have `$color-icon-secondary` below being referenced by the Icon styles rather than directly referencing `$color-secondary` or even `$color-telus-purple`.

https://github.com/telusdigital/tds-core/blob/48836eb97407ae9796142f05c8ecdb2e97f80883/shared/components/Icon/Icon.modules.scss#L7-L9

This change affects all consuming components anyway, but the only tangible impact is on the developer consuming these colours. Lastly, removing `$color-link` is a breaking change, and this package is being distributed.

Reply:
Hmmmm, I see this a little differently. I think the semantically named colours being used here are a relic of days past, when TDS was a giant CSS file, and before components, and this is something we should be moving away from. Taking the icon color example, there is one Icon component and all the knowledge of it should be contained therein.

By design, the Icon is in full control of its own styling, including colours, while the colours package defines the core colour palette. Any semantic naming should be done by the components themselves, not from the colours package, which would spread out that knowledge. If a colour has a specific purpose for a specific component, I think that component control it.

This hasn't come up until now, but this seems like a good conversation to have now. :)

Reply:
That all sounds agreeable to me. How about this:

- deprecate the verbose colour variables in `core-colours` using `@warn` (minor release)
- refactor consuming components to use the raw colour variables (patch releases)
- any new release of core-colours can be major

Thoughts?

Reply:
> deprecate the verbose colour variables in core-colours using @warn (minor release)

Deprecating Sass variables hasn't worked out well for us in the past. Consumers will see the warning every time their app compiles, with no way to turn it off, even if they are not using the variables. People complained ALOT, so we removed all Sass warnings.

Agree with the 2nd two steps.

Might be a bigger change than this story needs. So maybe for this one just focusing on the link is fine? And we can address the other stuff with a follow up story?

Reply:
Updated the PR with reverting `core-colours` and adding a colour code to the `Link` component. We'll just have to squash this PR to make sure only the `Link` component gets updated. :)